packages/fastify-renderer/src/node/renderers/react/ReactRenderer.tsx

```diff
@@ -69,14 +58,111 @@ export class ReactRenderer implements Renderer {
       for (const renderable of renderables) {
         await this.loadModule(this.entrypointRequirePathForServer(renderable))
       }
+
+      const mode = this.options.mode
+      this.transformHooks = this.plugin.hooks
+        .map(unthunk)
+        .map((hook) => hook.transform?.absolutePath)
+        .flatMap((hook) => (hook ? [hook] : []))
+
+      const modulePaths = await this.getPreloadPaths()
+      const paths = [...modulePaths, ...this.transformHooks]
+
+      this.workerPool = await createPool({
+        create() {
+          const workerData = {
+            paths,
+          }
+
+          let worker: Worker
+          switch (mode) {
+            case 'streaming':
+              worker = new Worker(require.resolve('./StreamingWorker.import.js'), {
+                workerData,
+              })
+            case 'sync':
+              worker = new Worker(require.resolve('./StaticWorker.import.js'), {
+                workerData,
+              })
+          }
+
+          return worker
+        },
+        async dispose(worker) {
+          await worker.terminate()
+        },
+      })
     }
   }
 
   /** The purpose of adding this function is to allow us to spy on this method, otherwise it isn't available in the class prototype */
   async render<Props>(render: Render<Props>): Promise<void> {
     return this.wrappedRender(render)
   }
+  private workerStreamRender<Props>(render: Render<Props>): NodeJS.ReadableStream {
+    const requirePath = this.entrypointRequirePathForServer(render)
+
+    const destination = this.stripBasePath(render.request.url, render.base)
+    if (!this.workerPool) throw new Error('WorkerPool not setup')
+    const passthrough = new PassThrough()
+
+    // Do not `await` or else it will not return
+    // until the whole stream is completed
+    void this.workerPool.use(
```

Comment:
SSR Stream -> Worker -> Postmessage -> Main thread -> Passthrough Stream -> document -> client

Reply:
This changed a bit but is still broadly the same. There is now a couple RenderBus instances in the mix.